### PR TITLE
Add DataFrame tests

### PR DESCRIPTION
**Pull request by zundertj**
Plugging a number of gaps in test coverage + adding some examples.

### CODE REVIEW

File: py-polars/polars/internals/frame.py [hunk 1]
```diff
@@ -2140,8 +2140,6 @@ def drop_nulls(self, subset: Optional[List[str]] = None) -> "DataFrame":
         └──────┴──────┘
 
         """
-        if subset is not None and isinstance(subset, str):
```
Comment (zundertj) [inline py-polars/polars/internals/frame.py:2143]: This is not allowed according to the type annotation, nor tested, so I removed it.

Comment (ritchie46) [reply py-polars/polars/internals/frame.py:2143]: This is so that we can call both:

`df.drop_nulls("foo")`

And `df.drop_nulls(["foo"])`

Comment (zundertj) [reply py-polars/polars/internals/frame.py:2143]: Okay, I was under the impression we would not want that anymore given there was no test and no type annotation for that case. I now added those.

**Comment by ritchie46** [comment]
I think this needs a rebase.

**Comment by zundertj** [comment]
Done.

**Comment by ritchie46** [comment]
:rocket:

**Pull request merged**